crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -1132,3 +1138,17 @@ export.validate_current_foreign_account
     # clean the stack
     dropw drop drop dropw
 end
+
+#! Hashes the map key.
+#!
+#! Inputs:  [KEY]
+#! Outputs: [HASHED_KEY]
+proc.hash_map_key
+  # pad four elements for the capacity of the hasher
+  # pad another four elements so the hash input is a multiple of 8
+  padw swapw padw swapw
+  # => [KEY, pad(8)]
+
+  hperm exec.rpo::squeeze_digest
+  # => [HASHED_KEY]
+end
```

Comment:
We could probably just use the `hash` [instruction](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/cryptographic_operations.html#hashing-and-merkle-trees).